src/test/java/com/qdesrame/openapi/test/BackwardCompatibilityTest.java

```diff
@@ -136,17 +137,51 @@ private void setMockMinimumValue(OpenAPI spec, int minValue) {
 
   @Test
   public void testEnumAdded() {
-    assertOpenApiBackwardIncompatible(OPENAPI_DOC8, OPENAPI_DOC9);
+    OpenAPI spec = OpenApiCompare.getOpenApiParser().read(OPENAPI_DOC8);
+    OpenAPI modSpec = OpenApiCompare.getOpenApiParser().read(OPENAPI_DOC8);
+    StringSchema status = getDogStatus(modSpec);
+    status.addEnumItem("zombie");
+    assertOpenApiBackwardIncompatible(spec, modSpec);
   }
 
   @Test
   public void testExtensibleEnumAdded() {
-    assertOpenApiBackwardCompatible(OPENAPI_DOC10, OPENAPI_DOC11, true);
+    OpenAPI spec = OpenApiCompare.getOpenApiParser().read(OPENAPI_DOC8);
+    StringSchema status = getDogStatus(spec);
+    status.addExtension("x-extensible-enum", status.getEnum());
+    status._enum(null);
+
+    OpenAPI modSpec = OpenApiCompare.getOpenApiParser().read(OPENAPI_DOC8);
+    StringSchema modStatus = getDogStatus(modSpec);
+    List<String> extendedEnum = modStatus.getEnum();
+    extendedEnum.add("zombie");
+    modStatus.addExtension("x-extensible-enum", extendedEnum);
+    modStatus._enum(null);
+    assertOpenApiBackwardCompatible(spec, modSpec, true);
+  }
+
+  private StringSchema getDogStatus(OpenAPI spec) {
+    return(StringSchema) spec.getComponents()
+            .getSchemas()
+            .get("Dog")
+            .getProperties()
+            .get("status");
   }
 
   @Test
   public void testExtensibleEnumRemoved() {
-    assertOpenApiBackwardIncompatible(OPENAPI_DOC11, OPENAPI_DOC10);
+    OpenAPI spec = OpenApiCompare.getOpenApiParser().read(OPENAPI_DOC8);
+    StringSchema status = getDogStatus(spec);
+    status.addExtension("x-extensible-enum", status.getEnum());
+    status._enum(null);
+
+    OpenAPI modSpec = OpenApiCompare.getOpenApiParser().read(OPENAPI_DOC8);
+    StringSchema modStatus = getDogStatus(modSpec);
+    List<String> extendedEnum = modStatus.getEnum();
+    extendedEnum.add("zombie");
```

Comment:
looks like you're still adding an enum, not removing one here

Reply:
They're reversed in the assertion, so the logic is correct, but it's a stupid way to test it. I'll just change the logic to remove and keep the assertion in the same order